lib/rake/thread_pool.rb

```diff
@@ -10,6 +9,7 @@ class ThreadPool # :nodoc: all
     # Creates a ThreadPool object.  The +thread_count+ parameter is the size
     # of the pool.
     def initialize(thread_count)
+      require "set"
```

Comment:
Honestly, looking at the usage the use of a set doesn't seem particularly useful. An Array would perform just I good I think.

The only operation that might be slower is `@threads.delete Thread.current`, but unless you have a huge amount of thread you probably won't notice any difference.

Reply:
I'm happy to change this to plain arrays too.